src/Illuminate/Console/Process/PendingProcess.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+namespace Illuminate\Console\Process;
+
+use Illuminate\Console\Process;
+use Illuminate\Console\Process\Results\Result;
+use Illuminate\Support\Traits\Macroable;
+
+class PendingProcess
+{
+    use Macroable;
+
+    /**
+     * The process's command.
+     *
+     * @var array<array-key, string>|string
+     */
+    protected $command = [];
+
+    /**
+     * Whether the process should be asynchronous.
+     *
+     * @var bool
+     */
+    protected $async = false;
+
+    /**
+     * The factory instance.
+     *
+     * @var \Illuminate\Console\Process\Factory
+     */
+    protected $factory;
+
+    /**
+     * The process's output callback.
+     *
+     * @var (callable(string, int): mixed)|null
+     */
+    protected $output;
+
+    /**
+     * The process's path.
+     *
+     * @var string|null
+     */
+    protected $path;
+
+    /**
+     * The process's timeout.
+     *
+     * @var float|null
+     */
+    protected $timeout = 60.0;
+
+    /**
+     * The stub callables that will handle processes.
+     *
+     * @var array<int, callable(\Illuminate\Console\Process): (\Illuminate\Console\Contracts\ProcessResult|null)>
+     */
+    protected $stubs = [];
+
+    /**
+     * The callbacks that should execute before the process starts.
+     *
+     * @var array<int, callable(\Illuminate\Console\Process): mixed>
+     */
+    protected $beforeStartCallbacks = [];
+
+    /**
+     * The callbacks that should execute after the process waits.
+     *
+     * @var array<int, callable(\Illuminate\Console\Process, \Illuminate\Console\Contracts\ProcessResult): mixed>
+     */
+    protected $afterWaitCallbacks = [];
+
+    /**
+     * Creates a new Pending Process instance.
+     *
+     * @param  \Illuminate\Console\Process\Factory  $factory
+     * @return void
+     */
+    public function __construct($factory)
+    {
+        $this->factory = $factory;
+    }
+
+    /**
+     * Toggle asynchronicity in the process..
+     *
+     * @param  bool  $async
+     * @return $this
+     */
+    public function async($async = true)
+    {
+        $this->async = $async;
+
+        return $this;
+    }
+
+    /**
+     * Dump the process and end the script before start the process.
+     *
+     * @return $this
+     */
+    public function dd()
+    {
+        return $this->beforeStart(fn ($process) => dd($process));
+    }
+
+    /**
+     * Dump the process before start the process.
+     *
+     * @return $this
+     */
+    public function dump()
+    {
+        return $this->beforeStart(fn ($process) => dump($process));
+    }
+
+    /**
+     * Register a stub callable that will intercept processes and be able to return stub process result.
+     *
+     * @param  array<int, callable(\Illuminate\Console\Process): (\Illuminate\Console\Contracts\ProcessResult|null)>  $callbacks
+     * @return $this
+     */
+    public function stubs($callbacks)
+    {
+        $this->stubs = $callbacks;
+
+        return $this;
```

Comment:
`return tap($this, fn () => $this->stubs = $callbacks);` to be consistent with the other setters/withers below?

Just for the record I prefer this style over the other, but as you already said you will keep the ones using `tap(...)` I guess it is better to be consistent, at least on the same class, don't you think?